libsolidity/ast/Types.cpp

```diff
@@ -617,6 +617,17 @@ FixedPointType::FixedPointType(unsigned _totalBits, unsigned _fractionalDigits,
 		"Invalid bit number(s) for fixed type: " +
 		dev::toString(_totalBits) + "x" + dev::toString(_fractionalDigits)
 	);
+
+	solAssert(
+		isValid(_totalBits, _fractionalDigits, isSigned()),
+		"Number of fractional digits do not fit the specified bit width"
+	);
+}
+
+bool FixedPointType::isValid(unsigned _totalBits, unsigned _fractionalDigits, bool _isSigned)
```

Comment:
Maybe this should also involve the lower and upper bounds for total bits, etc., used in the constructor of ``FixedPointType``? Having a function called ``isValid`` defined this way somewhat suggests that ``fixed512x1`` or ``fixed7x1`` would be valid... if this should still only check whether the fractional part fits into the bit width, maybe the function should be renamed accordingly?

Reply:
Added, though perhaps in #3389 the `TypeChecker` may have two helpers ("out of range" or "invalid configuration").